source/extensions/filters/http/gzip/config.cc

```diff
@@ -12,27 +12,14 @@ namespace Extensions {
 namespace HttpFilters {
 namespace Gzip {
 
-Server::Configuration::HttpFilterFactoryCb
-GzipFilterFactory::createFilter(const envoy::config::filter::http::gzip::v2::Gzip& proto_config) {
-  GzipFilterConfigSharedPtr config = std::make_shared<GzipFilterConfig>(proto_config);
-  return [config](Http::FilterChainFactoryCallbacks& callbacks) -> void {
-    callbacks.addStreamFilter(std::make_shared<GzipFilter>(config));
-  };
-}
-
-Server::Configuration::HttpFilterFactoryCb
-GzipFilterFactory::createFilterFactory(const Json::Object&, const std::string&,
-                                       Server::Configuration::FactoryContext&) {
-  NOT_IMPLEMENTED;
-}
-
 Server::Configuration::HttpFilterFactoryCb
 GzipFilterFactory::createFilterFactoryFromProto(const Protobuf::Message& proto_config,
```

Comment:
I would still argue for the typed vs. untyped variants and having `createFilterFactoryFromProto` defined in the superclass, with the typed variant here. This is cleaner, it saves basically a line of code and only exposes a typed interface to the subclass.

Reply:
Sorry I don't think I fully understood what you were asking for. I think I get it now. Will update.